setup.py

```diff
@@ -31,7 +31,9 @@
       author_email="jelmer@jelmer.uk",
       license="GNU GPLv3 or later",
       url="https://www.jelmer.uk/projects/xandikos",
-      install_requires=['icalendar', 'dulwich', 'defusedxml'],
+      install_requires=[
+          open('requirements.txt').readlines()
```

Comment:
TBH, for applications (eg: no libraries), I tend to prefer [this workflow](https://www.kennethreitz.org/essays/a-better-pip-workflow).

But it's really your call. `requirements.txt` is more of a de facto convention than anything else.